ruby-engine/lib/yggdrasil_engine.rb

```diff
@@ -85,6 +86,13 @@ def take_state(toggles)
     YggdrasilEngine.free_response(response_ptr)
   end
 
+  def get_state
```

Comment:
This looks subtly off, the Rust layer is returning a wrapped response that also includes a status code. That looks a little like this

``` json
{"status_code":"Ok","value":{"version":1,"features":[{"name":"testFeature","enabled":true,"strategies":[{"name":"default","variants":[]}]}],"query":null},"error_message":null}
```

You'd need to parse the JSON response and return the `value` here and/or make a decision around returning an error. Might be worth taking a peek at `get_variant` or `is_enabled` to see how they handle that dance